src/determinestrategy.jl

```diff
@@ -239,110 +239,110 @@ function determine_unroll_factor(
     roundpow2(max(1, round(Int, latency / (recip_throughput * num_reductions) ) ))
 end
 
-function tile_cost(X, U, T, UL, TL)
-    Tfactor = (num_iterations(TL, T)/TL)
-    Ufactor = (num_iterations(UL, U)/UL)
-    # X[1]*Tfactor*Ufactor + X[4] + X[2] * Tfactor + X[3] * Ufactor
-    X[1] + X[4] + X[2] * Tfactor + X[3] * Ufactor
+function unroll_cost(X, u₁, u₂, u₁L, u₂L)
+    u₂factor = (num_iterations(u₂L, u₂)/u₂L)
+    u₁factor = (num_iterations(u₁L, u₁)/u₁L)
+    # X[1]*u₂factor*u₁factor + X[4] + X[2] * u₂factor + X[3] * u₁factor
+    X[1] + X[4] + X[2] * u₂factor + X[3] * u₁factor
 end
```

Comment:
x[1] is the cost of operations that depend on u1loop and u2loop.
x[2] is the cost of operations that depend on u1loop.
x[3] is the cost of operations that depend on u2loop.
x[4] is the cost of operations that depend on neither.

By unrolling u2loop by a factor of u2, we get to reuse the results that only depend on u1loop u2 times, therefore we reduce the cost of the u1loop-dependent operations by roughly 1/u2.
Similar story for u1loop.
I don't recall why I didn't have `x[4]  * u₁factor * u₂factor`. I have to rethink that.
I think this is because operations didn't have a `reduced_children` when I first wrote this, so I just assumed the operations were being replicated in this part of the code.

Simply musing unrelated to this PR. I'll update it this and `solve_unroll` accordingly.